commit cc31f233cbb28b01ece0d581d0de1196993ba4fc
Author: xujian 30069781 <xujian141@h-partners.com>
Date:   Fri Mar 7 15:18:56 2025 +0800

    修复原仓库个别ENABLE_NLS选项不生效的问题，并关闭ENABLE_NLS去掉对GPL仓库 的依赖

diff --git a/glib/gi18n.h b/glib/gi18n.h
index dbb2cb3..9f1f8e9 100644
--- a/glib/gi18n.h
+++ b/glib/gi18n.h
@@ -22,13 +22,31 @@
 
 #include <glib.h>
 
-#include <libintl.h>
 #include <string.h>
 
+#ifdef ENABLE_NLS
+
+#include <libintl.h>
 #define  _(String) gettext (String)
 #define Q_(String) g_dpgettext (NULL, String, 0)
 #define N_(String) (String)
 #define C_(Context,String) g_dpgettext (NULL, Context "\004" String, strlen (Context) + 1)
 #define NC_(Context, String) (String)
 
+#else /* NLS is disabled */
+#define _(String) (String)
+#define Q_(String) (String)
+#define N_(String) (String)
+#define P_(String) (String)
+#define C_(Context,String) (String)
+#define NC_(Context, String) (String)
+#define textdomain(String) ((String) ? (String) : "messages")
+#define gettext(String) (String)
+#define dgettext(Domain,String) (String)
+#define dcgettext(Domain,String,Type) (String)
+#define dngettext(Domain,String1,String2,N) ((N) == 1 ? (String1) : (String2))
+#define bindtextdomain(Domain,Directory) (Domain) 
+#define bind_textdomain_codeset(Domain,Codeset)
+#endif
+
 #endif  /* __G_I18N_H__ */
diff --git a/meson.build b/meson.build
index 3773ce9..4b84f1c 100644
--- a/meson.build
+++ b/meson.build
@@ -244,7 +244,7 @@ glib_conf.set_quoted('PACKAGE_STRING', 'glib @0@'.format(meson.project_version()
 glib_conf.set_quoted('PACKAGE_TARNAME', 'glib')
 glib_conf.set_quoted('PACKAGE_URL', '')
 glib_conf.set_quoted('PACKAGE_VERSION', meson.project_version())
-glib_conf.set('ENABLE_NLS', 1)
+# glib_conf.set('ENABLE_NLS', 1)
 
 # used by the .rc.in files
 glibconfig_conf.set('LT_CURRENT_MINUS_AGE', soversion)
@@ -2150,10 +2150,8 @@ if libintl.found() and libintl.type_name() != 'internal'
   have_bind_textdomain_codeset = cc.has_function('bind_textdomain_codeset', dependencies: libintl_deps, prefix: libintl_prefix)
 else
   # using proxy-libintl fallback
-  libintl = dependency('intl', allow_fallback: true)
-  assert(libintl.type_name() == 'internal')
-  libintl_deps = [libintl]
-  have_bind_textdomain_codeset = true  # proxy-libintl supports it
+  libintl_deps = []
+  have_bind_textdomain_codeset = false  # proxy-libintl supports it
 endif
 
 glib_conf.set('HAVE_BIND_TEXTDOMAIN_CODESET', have_bind_textdomain_codeset)
diff --git a/meson_options.txt b/meson_options.txt
index 517d575..4ec2833 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -90,7 +90,7 @@ option('installed_tests',
 option('nls',
        type : 'feature',
        value : 'auto',
-       yield: true,
+       yield: false,
        description : 'Enable native language support (translations)')
 
 option('oss_fuzz',
