diff -Nura libdash-2.2/libdash/CMakeLists.txt patch/libdash/CMakeLists.txt
--- libdash-2.2/libdash/CMakeLists.txt	2013-04-03 02:40:41.000000000 -0700
+++ patch/libdash/CMakeLists.txt	2024-10-23 02:03:40.258801297 -0700
@@ -5,10 +5,12 @@
 include_directories(sdl/include)
 include_directories(zlib/include)
 include_directories(iconv/include)
+include_directories(libcurl/include)
 
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/bin)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/bin)
 set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/bin)
+set(CURL_LIBRARIES ${CMAKE_CURRENT_SOURCE_DIR}/libcurl/lib/libcurl.so)
 
 add_subdirectory(libdash)
-add_subdirectory(libdash_networkpart_test)
\ No newline at end of file
+add_subdirectory(libdash_networkpart_test)
